parquet-column/src/main/java/org/apache/parquet/internal/filter2/columnindex/RowRanges.java

```diff
@@ -75,16 +76,12 @@ long count() {
       return to - from + 1;
     }
 
-    @Override
-    public int compareTo(Range other) {
-      if (to < other.from) {
-        return -1;
-      } else if (from > other.to) {
-        return 1;
-      } else {
-        // Equality means the two ranges are overlapping
-        return 0;
-      }
+    boolean isBeforeThan(Range other) {
```

Comment:
(nit) These method names sound a bit strange in English. Some alternative you could consider:
- isBefore/isBehind ("than" is not needed and "behind" is better for non-temporal comparison than "after")
- precedes/follows